src/Symfony/Component/Message/DependencyInjection/MessagePass.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Message\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\Compiler\PriorityTaggedServiceTrait;
+use Symfony\Component\DependencyInjection\Compiler\ServiceLocatorTagPass;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\Message\Handler\ChainHandler;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class MessagePass implements CompilerPassInterface
+{
+    use PriorityTaggedServiceTrait;
+
+    private $messageBusService;
+    private $messageHandlerResolverService;
+    private $handlerTag;
+
+    public function __construct(string $messageBusService = 'message_bus', string $messageHandlerResolverService = 'message.handler_resolver', string $handlerTag = 'message_handler')
+    {
+        $this->messageBusService = $messageBusService;
+        $this->messageHandlerResolverService = $messageHandlerResolverService;
+        $this->handlerTag = $handlerTag;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition($this->messageBusService)) {
+            return;
+        }
+
+        $this->registerHandlers($container);
+    }
+
+    private function registerHandlers(ContainerBuilder $container)
+    {
+        $handlersByMessage = array();
+
+        foreach ($container->findTaggedServiceIds($this->handlerTag, true) as $serviceId => $tags) {
+            foreach ($tags as $tag) {
+                $handles = isset($tag['handles']) ? $tag['handles'] : $this->guessHandledClass($container, $serviceId);
+
+                if (!class_exists($handles)) {
+                    throw new RuntimeException(sprintf('The message class "%s" declared in `__invoke` function of service "%s" does not exist.', $handles, $serviceId));
```

Comment:
This message may not be very accurate in case you've defined the `handles` attribute, right?

Reply:
Yep, good point. Fixed in 9810a54006.